# **IPHONE SALES ANALYSIS PROJECT**


# iPhone Sales Analysis

This project focuses on analyzing sales data for iPhones, aiming to uncover patterns and insights related to pricing, customer preferences, and product performance. The dataset includes comprehensive details about various iPhone models and their market reception.

## Key Features of the Dataset

Product Name: Names of iPhone models.
Product URL: Links to product pages for reference.
Sale Price, MRP, and Discounted Price: Information about pricing and discounts.
Customer Feedback:
Number of Ratings: Total customer ratings.
Number of Reviews: Count of textual reviews.
Star Rating: Average star rating given by customers.
Technical Specifications:
RAM: Memory configurations for each model.
UPC: Universal Product Code for identification.
## Objectives

Identify pricing trends and the effectiveness of discounts.
Analyze customer feedback to gauge satisfaction and preferences.
Determine the impact of technical specifications on sales performance.
Provide actionable insights for marketing and product strategy.

In [ ]:
#importing required libraries

import pandas as pd
import numpy as np
import plotly.express as px
import plotly.graph_objects as go


In [ ]:
data= pd.read_csv("apple_products.csv")

In [ ]:
data

Product Name  \
0          APPLE iPhone 8 Plus (Gold, 64 GB)   
1   APPLE iPhone 8 Plus (Space Grey, 256 GB)   
2       APPLE iPhone 8 Plus (Silver, 256 GB)   
3            APPLE iPhone 8 (Silver, 256 GB)   
4              APPLE iPhone 8 (Gold, 256 GB)   
..                                       ...   
57            APPLE iPhone SE (Black, 64 GB)   
58           APPLE iPhone 11 (Purple, 64 GB)   
59            APPLE iPhone 11 (White, 64 GB)   
60            APPLE iPhone 11 (Black, 64 GB)   
61              APPLE iPhone 11 (Red, 64 GB)   

                                          Product URL  Brand  Sale Price  \
0   https://www.flipkart.com/apple-iphone-8-plus-g...  Apple       49900   
1   https://www.flipkart.com/apple-iphone-8-plus-s...  Apple       84900   
2   https://www.flipkart.com/apple-iphone-8-plus-s...  Apple       84900   
3   https://www.flipkart.com/apple-iphone-8-silver...  Apple       77000   
4   https://www.flipkart.com/apple-iphone-8-gold-2...  Apple       77000   
..                                                ...    ...         ...   
57  https://www.flipkart.com/apple-iphone-se-black...  Apple       29999   
58  https://www.flipkart.com/apple-iphone-11-purpl...  Apple       46999   
59  https://www.flipkart.com/apple-iphone-11-white...  Apple       46999   
60  https://www.flipkart.com/apple-iphone-11-black...  Apple       46999   
61  https://www.flipkart.com/apple-iphone-11-red-6...  Apple       46999   

      Mrp  Discount Percentage  Number Of Ratings  Number Of Reviews  \
0   49900                    0               3431                356   
1   84900                    0               3431                356   
2   84900                    0               3431                356   
3   77000                    0              11202                794   
4   77000                    0              11202                794   
..    ...                  ...                ...                ...   
57  39900                   24              95909               8161   
58  54900                   14              43470               3331   
59  54900                   14              43470               3331   
60  54900                   14              43470               3331   
61  54900                   14              43470               3331   

                 Upc  Star Rating   Ram  
0   MOBEXRGV7EHHTGUH          4.6  2 GB  
1   MOBEXRGVAC6TJT4F          4.6  2 GB  
2   MOBEXRGVGETABXWZ          4.6  2 GB  
3   MOBEXRGVMZWUHCBA          4.5  2 GB  
4   MOBEXRGVPK7PFEJZ          4.5  2 GB  
..               ...          ...   ...  
57  MOBFWQ6BR3MK7AUG          4.5  4 GB  
58  MOBFWQ6BTFFJKGKE          4.6  4 GB  
59  MOBFWQ6BVWVEH3XE          4.6  4 GB  
60  MOBFWQ6BXGJCEYNY          4.6  4 GB  
61  MOBFWQ6BYYV3FCU7          4.6  4 GB  

[62 rows x 11 columns]

In [ ]:
data.head() #prints top 5 rows of the table

Product Name  \
0         APPLE iPhone 8 Plus (Gold, 64 GB)   
1  APPLE iPhone 8 Plus (Space Grey, 256 GB)   
2      APPLE iPhone 8 Plus (Silver, 256 GB)   
3           APPLE iPhone 8 (Silver, 256 GB)   
4             APPLE iPhone 8 (Gold, 256 GB)   

                                         Product URL  Brand  Sale Price  \
0  https://www.flipkart.com/apple-iphone-8-plus-g...  Apple       49900   
1  https://www.flipkart.com/apple-iphone-8-plus-s...  Apple       84900   
2  https://www.flipkart.com/apple-iphone-8-plus-s...  Apple       84900   
3  https://www.flipkart.com/apple-iphone-8-silver...  Apple       77000   
4  https://www.flipkart.com/apple-iphone-8-gold-2...  Apple       77000   

     Mrp  Discount Percentage  Number Of Ratings  Number Of Reviews  \
0  49900                    0               3431                356   
1  84900                    0               3431                356   
2  84900                    0               3431                356   
3  77000                    0              11202                794   
4  77000                    0              11202                794   

                Upc  Star Rating   Ram  
0  MOBEXRGV7EHHTGUH          4.6  2 GB  
1  MOBEXRGVAC6TJT4F          4.6  2 GB  
2  MOBEXRGVGETABXWZ          4.6  2 GB  
3  MOBEXRGVMZWUHCBA          4.5  2 GB  
4  MOBEXRGVPK7PFEJZ          4.5  2 GB

In [ ]:
data.isnull().sum() #check for missing calues or null values

Product Name           0
Product URL            0
Brand                  0
Sale Price             0
Mrp                    0
Discount Percentage    0
Number Of Ratings      0
Number Of Reviews      0
Upc                    0
Star Rating            0
Ram                    0
dtype: int64

The above data shows that our data does not have any missing values.

In [ ]:
data.describe() #does basic statistics e.g., mean median min max etc

Sale Price            Mrp  Discount Percentage  Number Of Ratings  \
count      62.000000      62.000000            62.000000          62.000000   
mean    80073.887097   88058.064516             9.951613       22420.403226   
std     34310.446132   34728.825597             7.608079       33768.589550   
min     29999.000000   39900.000000             0.000000         542.000000   
25%     49900.000000   54900.000000             6.000000         740.000000   
50%     75900.000000   79900.000000            10.000000        2101.000000   
75%    117100.000000  120950.000000            14.000000       43470.000000   
max    140900.000000  149900.000000            29.000000       95909.000000   

       Number Of Reviews  Star Rating  
count          62.000000    62.000000  
mean         1861.677419     4.575806  
std          2855.883830     0.059190  
min            42.000000     4.500000  
25%            64.000000     4.500000  
50%           180.000000     4.600000  
75%          3331.000000     4.600000  
max          8161.000000     4.700000

# Question1: What are the top 10 highest-rated iPhones on Flipkart in India?

In [ ]:
data.head()

Product Name  \
0         APPLE iPhone 8 Plus (Gold, 64 GB)   
1  APPLE iPhone 8 Plus (Space Grey, 256 GB)   
2      APPLE iPhone 8 Plus (Silver, 256 GB)   
3           APPLE iPhone 8 (Silver, 256 GB)   
4             APPLE iPhone 8 (Gold, 256 GB)   

                                         Product URL  Brand  Sale Price  \
0  https://www.flipkart.com/apple-iphone-8-plus-g...  Apple       49900   
1  https://www.flipkart.com/apple-iphone-8-plus-s...  Apple       84900   
2  https://www.flipkart.com/apple-iphone-8-plus-s...  Apple       84900   
3  https://www.flipkart.com/apple-iphone-8-silver...  Apple       77000   
4  https://www.flipkart.com/apple-iphone-8-gold-2...  Apple       77000   

     Mrp  Discount Percentage  Number Of Ratings  Number Of Reviews  \
0  49900                    0               3431                356   
1  84900                    0               3431                356   
2  84900                    0               3431                356   
3  77000                    0              11202                794   
4  77000                    0              11202                794   

                Upc  Star Rating   Ram  
0  MOBEXRGV7EHHTGUH          4.6  2 GB  
1  MOBEXRGVAC6TJT4F          4.6  2 GB  
2  MOBEXRGVGETABXWZ          4.6  2 GB  
3  MOBEXRGVMZWUHCBA          4.5  2 GB  
4  MOBEXRGVPK7PFEJZ          4.5  2 GB

In [ ]:
high_rated= data.sort_values(by=["Star Rating"], ascending=False) #we get rating in descending order after ascending=False
high_rated= high_rated.head(10)
print(high_rated['Product Name'])

20     APPLE iPhone 11 Pro Max (Midnight Green, 64 GB)
17         APPLE iPhone 11 Pro Max (Space Grey, 64 GB)
16    APPLE iPhone 11 Pro Max (Midnight Green, 256 GB)
15               APPLE iPhone 11 Pro Max (Gold, 64 GB)
14              APPLE iPhone 11 Pro Max (Gold, 256 GB)
0                    APPLE iPhone 8 Plus (Gold, 64 GB)
29                     APPLE iPhone 12 (White, 128 GB)
32          APPLE iPhone 12 Pro Max (Graphite, 128 GB)
35                     APPLE iPhone 12 (Black, 128 GB)
36                      APPLE iPhone 12 (Blue, 128 GB)
Name: Product Name, dtype: object


In [ ]:
print(high_rated)

                                        Product Name  \
20   APPLE iPhone 11 Pro Max (Midnight Green, 64 GB)   
17       APPLE iPhone 11 Pro Max (Space Grey, 64 GB)   
16  APPLE iPhone 11 Pro Max (Midnight Green, 256 GB)   
15             APPLE iPhone 11 Pro Max (Gold, 64 GB)   
14            APPLE iPhone 11 Pro Max (Gold, 256 GB)   
0                  APPLE iPhone 8 Plus (Gold, 64 GB)   
29                   APPLE iPhone 12 (White, 128 GB)   
32        APPLE iPhone 12 Pro Max (Graphite, 128 GB)   
35                   APPLE iPhone 12 (Black, 128 GB)   
36                    APPLE iPhone 12 (Blue, 128 GB)   

                                          Product URL  Brand  Sale Price  \
20  https://www.flipkart.com/apple-iphone-11-pro-m...  Apple      117100   
17  https://www.flipkart.com/apple-iphone-11-pro-m...  Apple      117100   
16  https://www.flipkart.com/apple-iphone-11-pro-m...  Apple      131900   
15  https://www.flipkart.com/apple-iphone-11-pro-m...  Apple      117100   
14 

# Question2: How many ratings do the highest-rated iPhones on Flipkart have?

In [ ]:
data.head()

Product Name  \
0         APPLE iPhone 8 Plus (Gold, 64 GB)   
1  APPLE iPhone 8 Plus (Space Grey, 256 GB)   
2      APPLE iPhone 8 Plus (Silver, 256 GB)   
3           APPLE iPhone 8 (Silver, 256 GB)   
4             APPLE iPhone 8 (Gold, 256 GB)   

                                         Product URL  Brand  Sale Price  \
0  https://www.flipkart.com/apple-iphone-8-plus-g...  Apple       49900   
1  https://www.flipkart.com/apple-iphone-8-plus-s...  Apple       84900   
2  https://www.flipkart.com/apple-iphone-8-plus-s...  Apple       84900   
3  https://www.flipkart.com/apple-iphone-8-silver...  Apple       77000   
4  https://www.flipkart.com/apple-iphone-8-gold-2...  Apple       77000   

     Mrp  Discount Percentage  Number Of Ratings  Number Of Reviews  \
0  49900                    0               3431                356   
1  84900                    0               3431                356   
2  84900                    0               3431                356   
3  77000                    0              11202                794   
4  77000                    0              11202                794   

                Upc  Star Rating   Ram  
0  MOBEXRGV7EHHTGUH          4.6  2 GB  
1  MOBEXRGVAC6TJT4F          4.6  2 GB  
2  MOBEXRGVGETABXWZ          4.6  2 GB  
3  MOBEXRGVMZWUHCBA          4.5  2 GB  
4  MOBEXRGVPK7PFEJZ          4.5  2 GB

In [ ]:
#bar graph
iphones= high_rated['Product Name'].value_counts()
label= iphones.index
counts= high_rated["Number Of Ratings"]
fig= px.bar(high_rated, x=label, y=counts, title= "Number of ratings of highest rated iphones")
fig.show()


# Conclusion: The highest rated iphone 8 Plus(Gold, 64 GB) got 3431 ratings in total.

# Question3: Which iPhone has the highest number of reviews on Flipkart?

In [ ]:
data.head()

Product Name  \
0         APPLE iPhone 8 Plus (Gold, 64 GB)   
1  APPLE iPhone 8 Plus (Space Grey, 256 GB)   
2      APPLE iPhone 8 Plus (Silver, 256 GB)   
3           APPLE iPhone 8 (Silver, 256 GB)   
4             APPLE iPhone 8 (Gold, 256 GB)   

                                         Product URL  Brand  Sale Price  \
0  https://www.flipkart.com/apple-iphone-8-plus-g...  Apple       49900   
1  https://www.flipkart.com/apple-iphone-8-plus-s...  Apple       84900   
2  https://www.flipkart.com/apple-iphone-8-plus-s...  Apple       84900   
3  https://www.flipkart.com/apple-iphone-8-silver...  Apple       77000   
4  https://www.flipkart.com/apple-iphone-8-gold-2...  Apple       77000   

     Mrp  Discount Percentage  Number Of Ratings  Number Of Reviews  \
0  49900                    0               3431                356   
1  84900                    0               3431                356   
2  84900                    0               3431                356   
3  77000                    0              11202                794   
4  77000                    0              11202                794   

                Upc  Star Rating   Ram  
0  MOBEXRGV7EHHTGUH          4.6  2 GB  
1  MOBEXRGVAC6TJT4F          4.6  2 GB  
2  MOBEXRGVGETABXWZ          4.6  2 GB  
3  MOBEXRGVMZWUHCBA          4.5  2 GB  
4  MOBEXRGVPK7PFEJZ          4.5  2 GB

In [ ]:
#bar graph
iphones= high_rated['Product Name'].value_counts()
label= iphones.index
counts= high_rated["Number Of Reviews"]
fig= px.bar(high_rated, x=label, y=counts, title= "Number of reviews of highest rated iphones")
fig.show()


# Conclusion: The iphone 8 Plus(Gold,64GB) has the highest number of reviews(356) on flipkart.

# Question4: What is the relationship between the sale price of iPhones and the number of ratings on Flipkart?

In [ ]:
data.head()

Product Name  \
0         APPLE iPhone 8 Plus (Gold, 64 GB)   
1  APPLE iPhone 8 Plus (Space Grey, 256 GB)   
2      APPLE iPhone 8 Plus (Silver, 256 GB)   
3           APPLE iPhone 8 (Silver, 256 GB)   
4             APPLE iPhone 8 (Gold, 256 GB)   

                                         Product URL  Brand  Sale Price  \
0  https://www.flipkart.com/apple-iphone-8-plus-g...  Apple       49900   
1  https://www.flipkart.com/apple-iphone-8-plus-s...  Apple       84900   
2  https://www.flipkart.com/apple-iphone-8-plus-s...  Apple       84900   
3  https://www.flipkart.com/apple-iphone-8-silver...  Apple       77000   
4  https://www.flipkart.com/apple-iphone-8-gold-2...  Apple       77000   

     Mrp  Discount Percentage  Number Of Ratings  Number Of Reviews  \
0  49900                    0               3431                356   
1  84900                    0               3431                356   
2  84900                    0               3431                356   
3  77000                    0              11202                794   
4  77000                    0              11202                794   

                Upc  Star Rating   Ram  
0  MOBEXRGV7EHHTGUH          4.6  2 GB  
1  MOBEXRGVAC6TJT4F          4.6  2 GB  
2  MOBEXRGVGETABXWZ          4.6  2 GB  
3  MOBEXRGVMZWUHCBA          4.5  2 GB  
4  MOBEXRGVPK7PFEJZ          4.5  2 GB

In [ ]:
#scatter plot
figure = px.scatter(data_frame= data,
                    x="Number Of Ratings",
                    y="Sale Price",
                    size="Discount Percentage",
                    trendline="ols",
                    title="Relationship between Sale Price and Number of Ratings of iphones.")
figure.show()


# Conclusion: There is a negative linear relationship between the sale price and the number of ratings. i.e., iphones with lower sale price are being soled more.

# Question5: What is the relationship between the discount percentage and the number of ratings of iPhones on Flipkart?


In [ ]:
data.head()

Product Name  \
0         APPLE iPhone 8 Plus (Gold, 64 GB)   
1  APPLE iPhone 8 Plus (Space Grey, 256 GB)   
2      APPLE iPhone 8 Plus (Silver, 256 GB)   
3           APPLE iPhone 8 (Silver, 256 GB)   
4             APPLE iPhone 8 (Gold, 256 GB)   

                                         Product URL  Brand  Sale Price  \
0  https://www.flipkart.com/apple-iphone-8-plus-g...  Apple       49900   
1  https://www.flipkart.com/apple-iphone-8-plus-s...  Apple       84900   
2  https://www.flipkart.com/apple-iphone-8-plus-s...  Apple       84900   
3  https://www.flipkart.com/apple-iphone-8-silver...  Apple       77000   
4  https://www.flipkart.com/apple-iphone-8-gold-2...  Apple       77000   

     Mrp  Discount Percentage  Number Of Ratings  Number Of Reviews  \
0  49900                    0               3431                356   
1  84900                    0               3431                356   
2  84900                    0               3431                356   
3  77000                    0              11202                794   
4  77000                    0              11202                794   

                Upc  Star Rating   Ram  
0  MOBEXRGV7EHHTGUH          4.6  2 GB  
1  MOBEXRGVAC6TJT4F          4.6  2 GB  
2  MOBEXRGVGETABXWZ          4.6  2 GB  
3  MOBEXRGVMZWUHCBA          4.5  2 GB  
4  MOBEXRGVPK7PFEJZ          4.5  2 GB

In [ ]:
#scatter plot
figure=px.scatter(data_frame=data,
                  x="Number Of Ratings",
                  y="Discount Percentage",
                  size="Sale Price",
                  trendline="ols",
                  title="Relationship bw Discount Price and Number Of Ratings")

figure.show()

# Conclusion: There is a linear relationship between Discount Price and Number of Ratings.i.e., people buys those iphones which are having more discounted price.

# Question6: Can you figure out the least expensive and most expensive iPhones in the Indian market, alone with all their specifications?

In [ ]:
data.head()

Product Name  \
0         APPLE iPhone 8 Plus (Gold, 64 GB)   
1  APPLE iPhone 8 Plus (Space Grey, 256 GB)   
2      APPLE iPhone 8 Plus (Silver, 256 GB)   
3           APPLE iPhone 8 (Silver, 256 GB)   
4             APPLE iPhone 8 (Gold, 256 GB)   

                                         Product URL  Brand  Sale Price  \
0  https://www.flipkart.com/apple-iphone-8-plus-g...  Apple       49900   
1  https://www.flipkart.com/apple-iphone-8-plus-s...  Apple       84900   
2  https://www.flipkart.com/apple-iphone-8-plus-s...  Apple       84900   
3  https://www.flipkart.com/apple-iphone-8-silver...  Apple       77000   
4  https://www.flipkart.com/apple-iphone-8-gold-2...  Apple       77000   

     Mrp  Discount Percentage  Number Of Ratings  Number Of Reviews  \
0  49900                    0               3431                356   
1  84900                    0               3431                356   
2  84900                    0               3431                356   
3  77000                    0              11202                794   
4  77000                    0              11202                794   

                Upc  Star Rating   Ram  
0  MOBEXRGV7EHHTGUH          4.6  2 GB  
1  MOBEXRGVAC6TJT4F          4.6  2 GB  
2  MOBEXRGVGETABXWZ          4.6  2 GB  
3  MOBEXRGVMZWUHCBA          4.5  2 GB  
4  MOBEXRGVPK7PFEJZ          4.5  2 GB

In [ ]:
most_expensive= data.loc[data['Sale Price'].idxmax()]
least_expensive= data.loc[data['Sale Price'].idxmin()]
print("Most Expensive Iphone is: ", most_expensive)

print("\nLeast Expensive Iphone is: ",least_expensive)



Most Expensive Iphone is:  Product Name                        APPLE iPhone 12 Pro (Silver, 512 GB)
Product URL            https://www.flipkart.com/apple-iphone-12-pro-s...
Brand                                                              Apple
Sale Price                                                        140900
Mrp                                                               149900
Discount Percentage                                                    6
Number Of Ratings                                                    542
Number Of Reviews                                                     42
Upc                                                     MOBFWBYZ5UY6ZBVA
Star Rating                                                          4.5
Ram                                                                 4 GB
Name: 24, dtype: object

Least Expensive Iphone is:  Product Name                              APPLE iPhone SE (White, 64 GB)
Product URL            https://www.flipkart.

# Conclusion: The most expensive iphone is APPLE iPhone 12 Pro (Silver, 512 GB) and the least expensive iphone is APPLE iPhone SE (White, 64 GB) .